validator/src/validation/length.rs

```diff
@@ -1,42 +1,164 @@
-use crate::traits::HasLen;
+use std::{borrow::Cow, collections::{HashMap, HashSet, BTreeMap, BTreeSet}};
+
+#[cfg(feature = "indexmap")]
+use indexmap::{IndexMap, IndexSet};
```

Comment:
Thoughts on letting crates handle validator support themselves like they do for serde?